scripts/select_strains.py

```diff
@@ -206,28 +224,44 @@ def parse_metadata(segments, metadata_files):
     time_interval = determine_time_interval(args.time_interval, args.resolution)
 
     # derive additional lower inclusion date for "force-included strains"
-    reference_cutoff = date(year = time_interval[1].year - args.max_include_range, month=1, day=1)
+    lower_reference_cutoff = date(year = time_interval[1].year - args.max_include_range, month=1, day=1)
+    upper_reference_cutoff = time_interval[0]
 
     # read strains to exclude
     excluded_strains = read_strain_list(args.exclude) if args.exclude else []
     # read strains to include
     included_strains = read_strain_list(args.include) if args.include else []
 
+    # read in sequence names to determine which sequences already passed upstream filters
+    sequence_names_by_segment = parse_sequences(args.segments, args.sequences)
+
     # read in meta data, parse numeric dates
     metadata = parse_metadata(args.segments, args.metadata)
+
+    # eliminate all metadata entries that do not have sequences
+    filtered_metadata = {}
+    for segment in metadata:
+        filtered_metadata[segment] = {}
+        for name in metadata[segment]:
```

Comment:
since you have the names as sets already, you could use set intersections. but just a matter of style (and speed, but irrelevant here).